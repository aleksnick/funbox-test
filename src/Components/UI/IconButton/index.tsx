import React from "react";
import ICanStyled from "../../../Models/ICanStyled";
import { SimpleButtonProps, ButtonColor } from "../../UI/Button";
import UIIconButton from "@material-ui/core/IconButton";

export interface IconButtonProps extends SimpleButtonProps, ICanStyled {}

/**
 * Кнопка с иконкой
 *
 * @export
 * @class IconButton
 * @extends {React.Component<IconButtonProps>}
 */
export default class IconButton extends React.Component<IconButtonProps> {
  constructor(props: IconButtonProps) {
    super(props);
  }

  render() {
    const { className, color, href, disabled, children } = this.props;
    return (
      <UIIconButton
        className={className}
        color={color || ButtonColor.Default}
        disabled={!!disabled}
        href={href}
        onClick={this.onClick}
      >
        {children}
      </UIIconButton>
    );
  }

  onClick = (e: React.ChangeEvent<{}>) => {
    const onClick = this.props.onClick;
    if (onClick) {
      if (e) {
        e.preventDefault();
        e.stopPropagation();
      }
      onClick();
    }
  };
}
