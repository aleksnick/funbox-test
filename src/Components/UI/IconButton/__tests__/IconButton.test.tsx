import React from "react";
import IconButton from "../";
import { render, shallow } from "enzyme";

describe("IconButton", () => {
  describe("Snapshots", () => {
    test("Simple IconButton snapshot", () => {
      const wrapper = render(<IconButton>IconButton</IconButton>);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("Check properties", () => {
    test("IconButton OnClick", () => {
      let changeValue = false;
      const onClick = () => {
        changeValue = true;
      };

      const wrapper = shallow(
        <IconButton onClick={onClick}>IconButton</IconButton>
      );
      wrapper.simulate("click");

      expect(changeValue).toBe(true);
    });
  });
});
