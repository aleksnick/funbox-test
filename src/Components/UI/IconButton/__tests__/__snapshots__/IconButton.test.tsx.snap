// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton Snapshots Simple IconButton snapshot 1`] = `
<button
  class="MuiButtonBase-root-7 MuiIconButton-root-1"
  tabindex="0"
  type="button"
>
  <span
    class="MuiIconButton-label-6"
  >
    IconButton
  </span>
</button>
`;
